spec/graphql/union_type_spec.rb

```diff
@@ -153,7 +179,8 @@
     it "copies possible types without affecting the original" do
       union.possible_types # load the internal cache
       union_2 = union.dup
-      union_2.possible_types << type_3
```

Comment:
This changed some functionality, we no longer return the instance method from `possible_types` and instead return a copy of an array. Is this ok? Being able to mutate the possible types in the first place seems a bit concerning regardless.

To fix this properly it requires me storing a hash of context to the resulting possible types, OR when possible types is called without context then it returns the instance variable as before, but if called with a context returns the copy.

The latter is probably more reasonable and still keeps backwards compatibility if needed.

Reply:
Yes, I agree that the original usage in this test is a hack!